### PR TITLE
Set Helm release namespace to 'default'

### LINKED ISSUE #11: Certmanager Rolebinding is invalid
`demo-clusterrolebinding-gitlab-certmanager`

```
cannot sync resource: could not create object: ClusterRoleBinding.rbac.authorization.k8s.io "gitlab-certmanager" is invalid: subjects[0].namespace: Required value
```

```
    subjects:
    - kind: ServiceAccount
      name: gitlab-certmanager
      namespace: ""
```

Looks like we're passing through an empty namespace for the Certmanager service account at the moment; this will likely prevent Gitlab from working as expected.

### PR DESCRIPTION
**Pull request by negz**
This is a little confusing, as it only applies when chart templates reference the `.Release.Namespace` value, which most do not actually use to set their namespace.

Fixes #11

### CODE REVIEW

File: pkg/controller/gitlab/reconciler.go [hunk 1]
```diff
@@ -263,7 +263,9 @@ type helmResourceProducer struct {
 }
 
 func (p *helmResourceProducer) produce(v chartutil.Values) ([]*unstructured.Unstructured, error) {
-	templates, err := helm.Render(p.chartURL, helm.WithValues(v))
+	templates, err := helm.Render(p.chartURL,
+		helm.WithValues(v),
+		helm.WithReleaseNamespace(metav1.NamespaceDefault))
```
Comment (jbw976) [inline pkg/controller/gitlab/reconciler.go]: this is using the default namespace always, but can/should it be using the namespace from the gitlab CRD? or maybe not since this will be for a remote cluster?

Comment (negz) [reply pkg/controller/gitlab/reconciler.go]: There's a bunch of subtleties here. This isn't actually setting the namespace for everything; only cases in which a template refers to `{{ .Release.Namespace }}`. Most templates don't set their namespace using this value, instead they omit their namespace and rely on Tiller et al to specify it. See the comment and linked bug in `helm.go`.

The controller as currently implemented renders the template via Helm (without namespaces). AFAICT if we wanted to support deploying to a namespace beside `default` we'd need to set that namespace on each `*unstructured.Unstructured` template after the rendering phase.

So in short, everything is already deploying to `default`, this just ensures oddball cases like ObjectReferences in #11 also default to `default`.

Comment (jbw976) [reply pkg/controller/gitlab/reconciler.go]: do you think those details/context are worth adding as a comment here in the code? your call

**Comment by negz** [comment]
Please don't merge just yet - I'm still attempting to validate that this actually fixes #11.

File: pkg/controller/gitlab/reconciler.go (hunk 1 above)
Comment (negz) [reply pkg/controller/gitlab/reconciler.go]: I feel like they're pretty well documented already by my existing comment and link to the relevant bug in `helm.go`.

Comment (negz) [reply pkg/controller/gitlab/reconciler.go]: https://gitlab.com/charts/gitlab/blob/0a6cf11e70c1d77fe6cd6af419dacbe6e5c2dfa6/charts/gitlab/charts/unicorn/templates/deployment.yaml#L6

I note though that Gitlab have gone through and set their namespace explicitly, so we could set one if we wanted. We'd have to inject the namespace itself into the application, since I presume Gitlab doesn't create it.
